### PR TITLE
Add search plugin manager foundation

### PR DESCRIPTION
**Pull request by bbogdanov**

### CODE REVIEW

File: src/components/dialogs/searchDialog/PluginsManager.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    max-width="20rem"
+    scrollable
+  >
+    <v-card>
+      <v-card-title> <v-icon>mdi-toy-brick</v-icon> Plugins manager </v-card-title>
+      <v-card-text>
+        <v-switch
+          v-for="(plugin, key) in searchEngineState.searchPlugins"
+          :key="key"
+          :false-value="false"
```
Comment (CzBiX) [inline src/components/dialogs/searchDialog/PluginsManager.vue:13]: Does this required?

File: src/components/dialogs/searchDialog/PluginsManager.vue [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          :input-value="plugin.enabled"
+          :value="plugin.enabled"
```
Comment (CzBiX) [inline src/components/dialogs/searchDialog/PluginsManager.vue:15]: Why those lines?

File: src/store/index.ts [hunk 3]
```diff
@@ -18,7 +19,8 @@ const store = new Vuex.Store<RootState>({
     config: configStore,
     dialog: dialogStore,
     snackBar: snackBarStore,
-    addForm: addFormStore
+    addForm: addFormStore,
```
Comment (CzBiX) [inline src/store/index.ts:22]: That's why I always keep a comma at the array end if syntax allowed.

File: src/components/dialogs/searchDialog/PluginsManager.vue (hunk 2 above)
Comment (bbogdanov) [reply src/components/dialogs/searchDialog/PluginsManager.vue:15]: I believe you are asking here `why` I do not use `v-model`? 

`v-model` mutates the state. The reason for using these lines is that they continue to work as `v-model` but do not mutate the state. Just following one of the main concepts of redux pattern. (Only a reducer can mutate the state)

P.S. Actually when did this `switch` I found out we mutate the state that way without using the reducers. Based on it I understand I did mutate the state using `v-model` on several other places so I will have to change that.

File: src/components/dialogs/searchDialog/PluginsManager.vue (hunk 1 above)
Comment (bbogdanov) [reply src/components/dialogs/searchDialog/PluginsManager.vue:13]: `:false-value` by default sets `null`

File: src/components/dialogs/searchDialog/PluginsManager.vue (hunk 2 above)
Comment (CzBiX) [reply src/components/dialogs/searchDialog/PluginsManager.vue:15]: Just use `input-value`, remove `false-value` and `value` is enough.

Comment (bbogdanov) [reply src/components/dialogs/searchDialog/PluginsManager.vue]: You are right, I already do the handling in the actions. Good catch!

File: src/components/dialogs/searchDialog/PluginsManager.vue (hunk 1 above)
Comment (CzBiX) [reply src/components/dialogs/searchDialog/PluginsManager.vue]: Actually, set `false-value` isn't working, the change event value still null. I believe it's caused by `:value`

File: src/components/dialogs/searchDialog/PluginsManager.vue (hunk 2 above)
Comment (CzBiX) [reply src/components/dialogs/searchDialog/PluginsManager.vue]: And you will found the event value will be `false` now.

**Pull request merged**